packages/java-parser/src/tokens.js

```diff
@@ -199,12 +217,11 @@ createToken({
 // https://docs.oracle.com/javase/specs/jls/se11/html/jls-3.html#jls-3.10.4
 createToken({
   name: "CharLiteral",
-  pattern: /'(?:[^\\']|\\(?:(?:[btnfr"'\\/]|[0-7]|[0-7]{2}|[0-3][0-7]{2})|u[0-9a-fA-F]{4}))'/
+  pattern: MAKE_PATTERN("{{CharacterLiteral}}")
 });
 createToken({
   name: "StringLiteral",
-  // TODO: align with the spec, the pattern below is incorrect
-  pattern: /"[^"\\]*(\\.[^"\\]*)*"/
+  pattern: MAKE_PATTERN("{{StringLiteral}}")
```

Comment:
why do we need a stringLiteral fragment? can we not define that directly in the createToken?
as defined in: https://docs.oracle.com/javase/specs/jls/se11/html/jls-3.html#jls-StringLiteral

Reply:
We can, I was actually wondering whether to add this one FRAGMENT or not for aesthetic but let's go with your idea